jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientRedirectTest.java

```diff
@@ -292,24 +292,21 @@ protected void service(String target, Request jettyRequest, HttpServletRequest r
 
     @ParameterizedTest
     @ArgumentsSource(ScenarioProvider.class)
-    @Disabled
     public void testRedirectFailed(Scenario scenario) throws Exception
     {
-        // TODO this test is failing with timout after an ISP upgrade??  DNS dependent?
         start(scenario, new RedirectHandler());
 
-        try
-        {
-            client.newRequest("localhost", connector.getLocalPort())
+        ExecutionException e = assertThrows(ExecutionException.class,
+            () -> client.newRequest("localhost", connector.getLocalPort())
                 .scheme(scenario.getScheme())
                 .path("/303/doesNotExist/done")
                 .timeout(5, TimeUnit.SECONDS)
-                .send();
-        }
-        catch (ExecutionException x)
-        {
-            assertThat(x.getCause(), Matchers.instanceOf(UnresolvedAddressException.class));
-        }
+                .send());
+
+        assertThat("Cause", e.getCause(), Matchers.anyOf(
+            Matchers.instanceOf(UnresolvedAddressException.class),
+            Matchers.instanceOf(UnknownHostException.class))
```

Comment:
On my machine ...
On Java 8, the UnresolvedAddressException is returned.
On Java 11, and 14, UnknownHostException is what's returned.

Reply:
Let's see how this goes. 
I remember that redirecting to host `doesNotExist` was intercepted by some ISP and redirected to a "Want to buy this domain?" page and therefore succeeding the request rather than throwing an exception.

Reply:
We could implement DNS Hijacking detection easily enough in code.

How?

Generate 2 (or more) random domain names between 8 and 17 characters long and resolve them, if they have the same IP you have DNS Hijacking present.

Reply:
Go for it.

Reply:
DNS Hijacking detection added.